### PR TITLE
Added a Helm Chart compatibility column to "Version Compatibility Reference"

### PR DESCRIPTION
**Pull request by jwitz**
Airflow 2.0 requires your Airflow Helm Chart version to be at least 0.18.6. Because we didn't have a version requirement before, there was no place to document this info.

To fix this, I've added a new version column to the `Astronomer Certified` table that indicates the minimum required version for Airflow Helm Charts. I'm torn whether to say `N/A` or `Any` for the AC versions where it doesn't matter, let me know what you think!

### CODE REVIEW

**Review by paolaperaza — COMMENTED** [review]
@jwitz This looks good to me. TBH I don't know if there's a recommended Airflow chart for all other versions - will need someone else to take a look at this.

@schnie @kaxil Are you two the best folks to make a call?

**Comment by schnie** [comment]
`Any` seems fine to me for compatibility. We should always recommend the latest version when possible. Our single chart will support many versions of Airflow, until we start dropping support for AC versions. 

Say we drop AC 1.10.5 support today and release chart v1.0, removing some hacks we had in the chart to support it. At that point we would say AC 1.10.5 works with all versions of the chart up to v1.0. I don't think we've done anything like this yet, but could come up someday.

File: enterprise/v0.16/09_resources/03_version-compatibility-reference.md [hunk 1]
```diff
@@ -12,23 +12,23 @@ It's worth noting that while the tables below reference the minimum compatible v
...
 ## Astronomer Enterprise
 
-| Astronomer Platform  | Kubernetes        | Helm | Terraform | Postgres | Astronomer Certified                             | Python       |
-|----------------------|------------------|------|-----------|----------|--------------------------------------------------|---------------|
-| v0.16                | 1.16, 1.17, 1.18 | 3    | 0.12, 0.13.5      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14        | 3.6, 3.7, 3.8 |
-| v0.23 (*Coming Soon*)| 1.16, 1.17, 1.18 | 3    | 0.13.5      | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 2.0.0 | 3.6, 3.7, 3.8 |
+| Astronomer Platform   | Kubernetes       | Helm | Terraform    | Postgres | Astronomer Certified                             | Python        |
+| --------------------- | ---------------- | ---- | ------------ | -------- | ------------------------------------------------ | ------------- |
+| v0.16                 | 1.16, 1.17, 1.18 | 3    | 0.12, 0.13.5 | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14        | 3.6, 3.7, 3.8 |
+| v0.23 (_Coming Soon_) | 1.16, 1.17, 1.18 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 2.0.0 | 3.6, 3.7, 3.8 |
```
Comment (paolaperaza) [inline enterprise/v0.16/09_resources/03_version-compatibility-reference.md:18]:
```suggestion
| v0.23 | 1.16, 1.17, 1.18 | 3    | 0.13.5       | 9.6+     | 1.10.5, 1.10.7, 1.10.10, 1.10.12, 1.10.14, 2.0.0 | 3.6, 3.7, 3.8 |
```

**Review by paolaperaza — APPROVED** [review]
Thanks for the info @schnie. @jwitz I'll go ahead and give you an approval preemptively since the remaining changes are quick fixes. Thank you!

**Pull request merged**